near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs

```diff
@@ -111,49 +111,50 @@ impl ParsedVAA {
     }
 }
 
-sol! {
-    struct InitTransferWh {
-        address sender;
-        address tokenAddress;
-        uint128 nonce;
-        uint128 amount;
-        uint128 fee;
-        uint128 nativeFee;
-        string recipient;
-        string message;
-    }
+#[derive(Debug, BorshDeserialize)]
+struct DeployTokenWh {
+    token: String,
+    token_address: EvmAddress,
+}
 
-    struct FinTransferWh {
-        string token;
-        uint128 amount;
-        string recipient;
-        uint128 nonce;
-    }
+#[derive(Debug, BorshDeserialize)]
+struct FinTransferWh {
+    _token: String,
```

Comment:
lets add the type (first byte) to this struct and move the verification to the try_into.
see this https://github.com/Near-One/omni-bridge/pull/106#discussion_r1834088215

Reply:
I still like verifying inside `verify_vaa_callback`. We can forget and remove the check inside the conversion methods and introduce a vulnerability this way. However, I don't mind making this check in both places

Reply:
50f55ccb6f04ec9987a25384a537516cda4dd51a